modules/umig/variables.tf

```diff
@@ -47,10 +47,11 @@ variable "num_instances" {
 
 variable "named_ports" {
   description = "Named name and named port"
-  type        = "list"
+  type        = list(map(string))
```

Comment:
This should be typed as `list(map(number))`.

Reply:
IMHO it is not the right way, because as I see https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_compute_instance_group.go#L55-L71
Based on this, we have a heterogeneous structure with different types of data, and it is better to use an object, it should be something like:
variable "named_ports" {
  description = "Named name and named port"
  type = list(object({
    name = string
    port = number
  }))
  default = []
}
I'll do appropriate changes and test it now

Reply:
![image](https://user-images.githubusercontent.com/2316547/61838744-6934bc00-ae93-11e9-86a4-59fe2fb7082d.png)
![image](https://user-images.githubusercontent.com/2316547/61838772-95503d00-ae93-11e9-9a6e-cf5323900c71.png)
![image](https://user-images.githubusercontent.com/2316547/61838804-b4e76580-ae93-11e9-9108-aee701151136.png)
![image](https://user-images.githubusercontent.com/2316547/61838837-d1839d80-ae93-11e9-8f6d-7f954d3576c9.png)
![image](https://user-images.githubusercontent.com/2316547/61838874-fc6df180-ae93-11e9-8d85-67a512261dad.png)